### PR TITLE
fix: Display error message on 2FA modal.

### LINKED ISSUE #2543: Login modal missing 2fa error
Reproduction steps
1. Use the login modal that pops up for the comment login.

image

2. Enter an incorrect 2FA code.

3. The user is not logged in, but the login modal disappears as if the login was succesful.  There is not error anywhere on the page to indicate that the 2fa code was incorrect.

The expected behavior is that the login modal displayes the 2fa error code that is displayed on the normal login screen (the picture shown below) then allows the user to re-attempt to login.

image

### PR DESCRIPTION
**Pull request by victorgcramos**
Close #2543

This PR adds error messages on Two-Factor Authentication modal verification.

### UI Changes Screenshot

Before:
2543-2fa-verify-before

After:
2543-2fa-verify-after

### CODE REVIEW

**Review by lukebp — COMMENTED** [review]
tACK

Are you able to add tests for this?

**Comment by lukebp** [comment]
I like how the error message is displayed in the 2fa modal instead of closing the modal out, which is what is done on the normal login screen.

Is this something that we can add to the normal login workflow as well? Not in this PR, but a separate one.

**Comment by victorgcramos** [comment]
@lukebp Thanks for the review, mate! Since there are two PRs regarding the 2FA, I'll wait for them to get merged and include the tests for both cases. I'll also include the Login error handling on another PR.

File: src/components/ModalTotpVerify.jsx [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
 
 const ModalTotpVerify = ({ show, onClose, onVerify }) => {
+  const [error, setError] = useState();
   const handleChange = (v) => {
     if (v.length === TOTP_CODE_LENGTH) {
-      onVerify(v).then(() => onClose());
+      onVerify(v)
+        .then(() => {
+          onClose();
+          // setError();
```
Comment (tiagoalvesdulce) [inline src/components/ModalTotpVerify.jsx:14]: I think this can go, right?

**Review by tiagoalvesdulce — COMMENTED** [review]
LGTM, left a small comment.

File: src/components/ModalTotpVerify.jsx (hunk 1 above)
Comment (victorgcramos) [reply src/components/ModalTotpVerify.jsx:14]: sure! My bad

**Comment by lukebp** [comment]
This looks ready to merge.  Are we waiting on anything else?

**Comment by victorgcramos** [comment]
I guess so. Needs final approval from @tiagoalvesdulce.

**Review by tiagoalvesdulce — APPROVED** [review]
ACK

**Pull request merged**